### PR TITLE
Fix: `max-len`: `ignoreTemplateLiterals`: handle 3+ lines (fixes #7125)

### LINKED ISSUE #7125: `max-len` rule `ignoreTemplateLiterals` not ignore multi-line template literals
**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** 6.2.1
* **npm Version:** 3.9.6

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-eslint

**Please show your full configuration:** Isolated config and used only one rule.
```yaml
        "max-len": [
            2,
            100, {
                "ignoreComments": true,
                "ignoreTrailingComments": true,
                "ignoreUrls": true,
                "ignoreStrings": true,
                "ignoreTemplateLiterals": true
            }
        ]
```

**What did you do? Please include the actual source code causing the issue.**
```js
        this.x = `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
        aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
        aaaaaaaaaaaaa`;
```
untitled

**What did you expect to happen?** I expected eslint to **not** throw a `max-len` error as I have **ignoreTemplateLiterals** turned on.

**What actually happened? Please include the actual, raw output from ESLint.** It threw an error for multi-line template literal.

/cc #5805 and #7049.

### PR DESCRIPTION
**Pull request by ljharb**
**What is the purpose of this pull request? (put an "X" next to item)**

```
[ ] Documentation update
[x] Bug fix ([template](https://github.com/eslint/eslint/blob/master/templates/bug-report.md))
[ ] New rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:
```

*If the item you've check above has a template, please paste the template questions here and answer them.*
See https://github.com/eslint/eslint/issues/7125 for template answers

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [x] I've included tests for my change
- [x] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**
In `groupByLineNumber`, instead of assuming that "start" and "end" are next to each other, add the entire range of lines to the group.

**Is there anything you'd like reviewers to focus on?**
¯\\\_(ツ)_/¯

### CODE REVIEW

**Comment by bgw** [comment]
:+1:  LGTM

**Comment by platinumazure** [comment]
Could you add a test with line continuations such that the resulting string is at least three lines long? Otherwise LGTM.

**Comment by btmills** [comment]
Other than @platinumazure's continuation suggestion, LGTM.

**Comment by ljharb** [comment]
@platinumazure is that not what the test I added does?

**Comment by platinumazure** [comment]
@ljharb Apologies for being unclear. I mean a 3-line version of [this test](https://github.com/eslint/eslint/blob/master/tests/lib/rules/max-len.js#L99-L102), where the middle line exceeds the max-len limit. Make sense?

**Comment by btmills** [comment]
@platinumazure to confirm my understanding, you want to keep the existing test and add one using continuations, so we cover both of the following?

```js
const foo = `This template literal is very
long and spans multiple lines. Its output will
preserve line breaks.`;
const bar = `This template literal is also very \
long and spans multiple lines. However, its output will \
be combined into one long line.`;
```

Like that?

**Comment by ljharb** [comment]
I read it as asking for a test where the not-first-or-last line is the one that's too long?

**Comment by platinumazure** [comment]
@ljharb

> I read it as asking for a test where the not-first-or-last line is the one that's too long?

Yes, but for both template literals and string literals. You have added one for template literals, but I wanted one for a string literal.

When you implemented the ignoreStrings/ignoreTemplateLiterals option originally, you very kindly added tests for both a 2-line template literal and a 2-line string (with one backslash line continuation). Now we are adding tests for a 3-line template literal. I just want a 3-line string test 9with two backslash line continuations) just to be safe.

In other words, long story short, @btmills' [post](https://github.com/eslint/eslint/pull/7138#issuecomment-246729770) describes exactly what I'm looking for. I used "continuation" to refer to the backslash-before-newline construct, which I've heard of as a "line continuation" in languages like C-- I do apologize if I could have been more clear.

Hope that makes sense.

**Comment by btmills** [comment]
@platinumazure should my second (`const bar = ...`) example use double quotes instead of backticks then?

**Comment by btmills** [comment]
Also, good call on the continuations. I wouldn't have thought about that.

**Comment by platinumazure** [comment]
Ah, yeah, I was thinking double or single quotes in the second example. I didn't know template literals also allowed line continuation sequences and combined into one line, though. Maybe having three test cases wouldn't hurt.

**Comment by btmills** [comment]
Cool! LGTM

**Comment by platinumazure** [comment]
LGTM, thanks for bearing with me @ljharb!

**Comment by platinumazure** [comment]
I'll merge this tonight unless someone beats me to it.

**Comment by nzakas** [comment]
@platinumazure PRs must be open for at least 48 hours before merging, so this one needs to be left open for another 36 hours. :)

**Comment by platinumazure** [comment]
Oops, I was thinking of docs-only changes. For some reason I thought bugfixes applied too.

**Comment by nzakas** [comment]
You're confusing who can merge PRs with how long to wait. :) Committers can merge docs changes and bug fixes, but there's a waiting period for all code PRs to ensure we're able to get more eyes on it. See http://eslint.org/docs/maintainer-guide/pullrequests.html#when-to-merge-a-pull-request

**Comment by ljharb** [comment]
This has now been up for 60 hours, is it good to merge? :-)

**Comment by btmills** [comment]
Huh, that's weird. I approved, and all checks are green, but merge is still red. Could be a bug, I'll give it a minute to see if it clears up. Maybe restart a Travis build to give it a kick.

**Comment by ljharb** [comment]
@btmills i still see it looking grey, but maybe it's green for you?

**Comment by btmills** [comment]
I'm going to go ahead and merge this since all of our status checks are passing individually. Hopefully the status check problem is a fluke and doesn't repeat.
